### PR TITLE
Improve threads aborting mechanism at Emu.Stop()

### PR DESCRIPTION
**Pull request by elad335**
* Improve threads aborting mechanism at emulation stopping.

### CODE REVIEW

File: rpcs3/Emu/System.cpp [hunk 1]
```diff
@@ -1534,13 +1534,7 @@ void Emulator::Stop(bool restart)
...
 	});
 
-	if (auto rsx = g_fxo->try_get<rsx::thread>())
-	{
-		// TODO: notify?
-		rsx->state += cpu_flag::exit;
-	}
-
-	cpu_thread::stop_all();
+	id_manager::g_mutex.lock_unlock();
```
Comment (Nekotekina) [inline rpcs3/Emu/System.cpp:1537]: What is this for? Also please add log notice message after signaling the threads.

**Review by Nekotekina — CHANGES_REQUESTED** [review]

File: rpcs3/Emu/System.cpp (hunk 1 above)
Comment (elad335) [reply rpcs3/Emu/System.cpp:1537]: In coordination with cpu_thread constructor which checks if emulation is stopped. (constructor assumes operates under IDM mutex) This prevents a race where newly created cpu_thread would not be stopped.

Comment (elad335) [reply rpcs3/Emu/System.cpp]: Added log message.

**Review by Nekotekina — APPROVED** [review]

**Comment by Megamouse** [comment]
Well. now "Stopping the Emulator took too long" is happening all the time

**Pull request merged**